sci-libs/vtk/vtk-9.1.0.ebuild

```diff
@@ -0,0 +1,601 @@
+		-DVTK_BUILD_ALL_MODULES=$(usex all-modules ON OFF)
+		# we use the pre-built documentation and install these with USE=doc
+		-DVTK_BUILD_DOCUMENTATION=OFF
+		-DVTK_BUILD_EXAMPLES=$(usex examples ON OFF)
+
+		-DVTK_ENABLE_KITS=ON
+		# defaults to ON: USE flag for this?
+		-DVTK_ENABLE_REMOTE_MODULES=OFF
+
+#		-DVTK_DATA_STORE="${S}/.ExternalData"
+
+		-DVTK_DEFAULT_RENDER_WINDOW_OFFSCREEN=ON
+
+		-DVTK_GROUP_ENABLE_Imaging=$(usex imaging "YES" "DONT_WANT")
+		-DVTK_GROUP_ENABLE_Qt=$(usex qt5 "YES" "DONT_WANT")
+		-DVTK_GROUP_ENABLE_Rendering=$(usex rendering "YES" "DONT_WANT")
+		-DVTK_GROUP_ENABLE_StandAlone="YES"
+		-DVTK_GROUP_ENABLE_Views=$(usex views "YES" "DONT_WANT")
+		-DVTK_GROUP_ENABLE_Web=$(usex web "YES" "DONT_WANT")
+
+		-DVTK_INSTALL_SDK=ON
+
+		-DVTK_MODULE_ENABLE_VTK_vtkm="WANT"
+		-DVTK_MODULE_ENABLE_VTK_IOGeoJSON="WANT"
+
+		# TODO: update one cgnslib-4.1.1 is packaged
+		-DVTK_MODULE_USE_EXTERNAL_VTK_cgns=OFF
+		# available in ::guru, so avoid  detection if installed
+#		-DVTK_MODULE_USE_EXTERNAL_VTK_cli11=OFF
+		# these are not package in Gentoo
+		-DVTK_MODULE_USE_EXTERNAL_VTK_exprtk=OFF
+		# TODO: update once libfmt-8.1.0 has been packaged
+		-DVTK_MODULE_USE_EXTERNAL_VTK_fmt=OFF
+		# not pacakged in Gentoo
+		-DVTK_MODULE_USE_EXTERNAL_VTK_ioss=OFF
+		# available in ::guru, so avoid detection if installed
+#		-DVTK_MODULE_USE_EXTERNAL_VTK_zfp=OFF
+
+		# FIXME: this is the default
+#		-DVTK_PYTHON_VERSION="3"
+
+		-DVTK_RELOCATABLE_INSTALL=ON
+
+		-DVTK_USE_CUDA=$(usex cuda ON OFF)
+		# use system libraries where possible
+		-DVTK_USE_EXTERNAL=ON
+		-DVTK_USE_MPI=$(usex mpi ON OFF)
+		-DVTK_USE_TK=$(usex tk ON OFF)
+
+		# FIXME: do we still want this?
+#		-DVTK_VERSIONED_INSTALL=ON
+
+		-DVTK_WRAP_JAVA=$(usex java ON OFF)
+		-DVTK_WRAP_PYTHON=$(usex python ON OFF)
+	)
+
+	if use all-modules; then
+		mycmakeargs+=(
+			-DVTK_ENABLE_OSPRAY=OFF
+			# TODO: most of these are tied to the VTK_ENABLE_REMOTE_MODULES
+			# option. Check whether we can download them clean and enable
+			# them.
+			-DVTK_MODULE_ENABLE_VTK_DomainsMicroscopy="DONT_WANT"
+			-DVTK_MODULE_ENABLE_VTK_FiltersOpenTURNS="DONT_WANT"
+			-DVTK_MODULE_ENABLE_VTK_IOADIOS2="DONT_WANT"
+			-DVTK_MODULE_ENABLE_VTK_IOPDAL="DONT_WANT"
+			-DVTK_MODULE_ENABLE_VTK_MomentInvariants="DONT_WANT"
+			-DVTK_MODULE_ENABLE_VTK_PoissonReconstruction="DONT_WANT"
+			-DVTK_MODULE_ENABLE_VTK_Powercrust="DONT_WANT"
+			-DVTK_MODULE_ENABLE_VTK_RenderingOpenVR="DONT_WANT"
+			-DVTK_MODULE_ENABLE_VTK_SignedTensor="DONT_WANT"
+			-DVTK_MODULE_ENABLE_VTK_SplineDrivenImageSlicer="DONT_WANT"
+			-DVTK_MODULE_ENABLE_VTK_vtkDICOM="DONT_WANT"
+			-DVTK_MODULE_USE_EXTERNAL_vtkkissfft=ON
+		)
+	fi
+
+	# TODO: consider removing USE flags and enable by default
+	if use boost; then
+		mycmakeargs+=(
+			-DVTK_MODULE_ENABLE_VTK_InfovisBoost="WANT"
+			-DVTK_MODULE_ENABLE_VTK_InfovisBoostGraphAlgorithms="WANT"
+		)
+	fi
+
+	if use cuda; then
+		local cuda_arch=
+		case ${VTK_CUDA_ARCH:-native} in
```

Comment:
I guess we may want to add a comment here explaining how to update this in future

Reply:
You mean, like when NVidia adds new architectures and/or obsoletes old arches? Good idea!